### PR TITLE
connect: fix leaf cert generation with RSA keys

### PR DESCRIPTION
**Pull request by tradel**
Fixes the case where CA is configured to use RSA keys as in #6055, but leaf cert generation was still hardcoded to use ECDSA keys and `x509.ECDSAWithSHA256` as its signature algorithm.

### CODE REVIEW

File: agent/testagent.go [hunk 1]
```diff
@@ -353,6 +353,8 @@ func TestConfig(sources ...config.Source) *config.RuntimeConfig {
 				enabled = true
 				ca_config {
 					cluster_id = "` + connect.TestClusterID + `"
+                	private_key_type = "ec"
```
Comment (banks) [inline agent/testagent.go]: nit: is there a whitespace issue here or just GitHub having a bad day?

Comment (tradel) [reply agent/testagent.go]: It passed `go fmt` so I'm assuming it's Github being weird.

**Comment by tradel** [comment]
> Is there some reason I'm forgetting that it's hard to re-use some existing tests with RSA to be sure it works?

I wrote a bunch of tests in #6055 that create certs with both `ec` and `rsa` private keys in various bit lengths. This is more of a bugfix.

I suppose it should be possible to write a test that starts up a server, generates a leaf cert, and then confirms it has the right PK type. Would you like me to do that?

**Comment by banks** [comment]
I spent a while going over this and realised that the `CARoot.PrivateKeyType` field is being populated from the _config_ not from the actual key used. That means it's usually zero which is not helpful, but even worse if you use an externally created CA key/cert (possible with both Vault and Consul providers currently) but configured something different in Consul, we'd end up showing the wrong thing entirely.

I originally thought there was an additional issue here which I was mistaken about but by the time I realised I'd started re-working this in a new branch which had a lot of extra tests so I'm going to close this out in favor of the new PR: #6638.

Thanks for work here Todd it was mostly re-used in some form!

**Pull request closed without merging**